### PR TITLE
Allow users to be removed from systems

### PR DESCRIPTION
**Pull request by mulby**
This will allow us to keep a list of users that are allowed explicitly not
allowed to access systems.

```
user_info:
 - name: someonewholeft
   state: absent
 - name: activeaccount
   type: admin
   github: true
```

The above settings would ensure that the user named "someonewholeft" is not
able to access the system and "activeaccount" is.

### CODE REVIEW

**Comment by mulby** [comment]
@e0d @jarv not sure if you want this, but I made these modifications on our fork of this role

**Comment by feanil** [comment]
looks good to me.

File: playbooks/roles/user/tasks/main.yml [hunk 1]
```diff
@@ -74,6 +74,12 @@
 - name: create the edxadmin group
   group: name=edxadmin state=present
 
+- name: ensure sudoers.d is read
```
Comment (jarv) [inline playbooks/roles/user/tasks/main.yml]: why is this necessary? afaik reading sudoers.d is the default.

File: playbooks/roles/user/tasks/main.yml [hunk 2]
```diff
@@ -85,19 +91,21 @@
...
 - name: create .ssh directory
   file:
     path=/home/{{ item.name }}/.ssh state=directory mode=0750
     owner={{ item.name }}
+  when: item.get('state', 'present') == 'present'
   with_items: user_info
 
 - name: assign admin role to admin users
-  user:
+  user: >
```
Comment (jarv) [inline playbooks/roles/user/tasks/main.yml:106]: I would go ahead and fix the other tasks as well.

File: playbooks/roles/user/tasks/main.yml (hunk 1 above)
Comment (mulby) [inline playbooks/roles/user/tasks/main.yml]: Unfortunately on the EMR master nodes it is not the default. AFAICT they are actually running some version of Debian instead of LTS ubuntu. They only ship with python 2.6 as well... its frustrating.

If you do decide to merge this we should probably test it on the boxes you typically run against to make sure the regex matches etc.

Comment (jarv) [inline playbooks/roles/user/tasks/main.yml]: Can you provision a sandbox using this branch then and add a comment saying why it's necessary?

File: playbooks/roles/user/tasks/main.yml (hunk 2 above)
Comment (mulby) [inline playbooks/roles/user/tasks/main.yml:106]: Ah, I didn't notice the other ones.

Comment (mulby) [inline playbooks/roles/user/tasks/main.yml:106]: Done.

File: playbooks/roles/user/tasks/main.yml (hunk 1 above)
Comment (mulby) [inline playbooks/roles/user/tasks/main.yml]: Sure thing, firing one up now.

**Comment by mulby** [comment]
Sandbox deploy succeeded: http://jenkins.edx.org:8080/view/Ansible/job/ansible-provision/118/console

**Comment by feanil** [comment]
:rocket:

**Comment by jarv** [comment]
@mulby did you ssh in and verify you can run sudo? if so :+1:

**Comment by mulby** [comment]
Yeah I tested sshing in and using sudo, both worked. What's up with travis?

**Comment by jarv** [comment]
maybe rebase? The .travis file was fixed and merged into master today
